src/main/java/org/dataloader/DataLoaderHelper.java

```diff
@@ -292,78 +298,91 @@ private void possiblyClearCacheEntriesOnExceptions(List<K> keys) {
             return futureCache.get(cacheKey);
         }
 
-        /*
-        We haven't been asked for this key yet. We want to do one of two things:
-
-        1. Check if our cache store has it. If so:
-            a. Get the value from the cache store
-            b. Add a recovery case so we queue the load if fetching from cache store fails
-            c. Put that future in our futureCache to hit the early return next time
-            d. Return the resilient future
-        2. If not in value cache:
-            a. queue or invoke the load
-            b. Add a success handler to store the result in the cache store
-            c. Return the result
-         */
-        final CompletableFuture<V> future = new CompletableFuture<>();
-
-        valueCache.get(cacheKey).whenComplete((cachedValue, getCallEx) -> {
-            if (getCallEx == null) {
-                future.complete(cachedValue);
-            } else {
-                synchronized (dataLoader) {
-                    queueOrInvokeLoader(key, loadContext, batchingEnabled)
-                            .whenComplete(setValueIntoCacheAndCompleteFuture(cacheKey, future));
-                }
-            }
-        });
```

Comment:
This was the old async code in respect to the `load` call.  This has been removed and cache lookup is now done in batch inside the "batch load" call done because of a call to `dispatch`